modules/javafx.graphics/src/main/java/com/sun/glass/ui/win/WinRobot.java

```diff
@@ -24,32 +24,96 @@
  */
 package com.sun.glass.ui.win;
 
-import com.sun.glass.ui.*;
-import java.nio.IntBuffer;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.MouseButton;
+import javafx.scene.paint.Color;
+
+import com.sun.glass.ui.Application;
+import com.sun.glass.ui.GlassRobot;
 
 /**
  * MS Windows platform implementation class for Robot.
  */
-final class WinRobot extends Robot {
+final class WinRobot extends GlassRobot {
 
-    @Override protected void _create() {
+    @Override
+    public void create() {
         // no-op
     }
-    @Override protected void _destroy() {
+
+    @Override
+    public void destroy() {
         // no-op
     }
 
-    @Override native protected void _keyPress(int code);
-    @Override native protected void _keyRelease(int code);
+    native protected void _keyPress(int code);
+    @Override
+    public void keyPress(KeyCode code) {
+        Application.checkEventThread();
+        _keyPress(code.getCode());
+    }
+
+    native protected void _keyRelease(int code);
+    @Override
+    public void keyRelease(KeyCode code) {
+        Application.checkEventThread();
+        _keyRelease(code.getCode());
+    }
+
 
-    @Override native protected void _mouseMove(int x, int y);
-    @Override native protected void _mousePress(int buttons);
-    @Override native protected void _mouseRelease(int buttons);
-    @Override native protected void _mouseWheel(int wheelAmt);
+    native protected void _mouseMove(int x, int y);
+    @Override
+    public void mouseMove(double x, double y) {
+        Application.checkEventThread();
+        _mouseMove((int) x, (int) y);
```

Comment:
Just curiously, can we lose a precision here(and in other places). For example if the scale of the screen=2 and x=0.5, where we should place the mouse? at "0" pixel or at "1" pixel?

Reply:
This is a good question. I would recommend filing a follow-up JBS issue to look at all of the newly added conversions from double to int. We should probably pass double/float values down to the native code.